add `index.d.ts` to the project

Currently, the JS/TS language server reports the following error: `Cannot find module 'get-arduino-tools' or its corresponding type declarations.ts(2307)`

It should be available `types` from the `package.json`

--- a/package.json
+++ b/package.json
@@ -20,20 +20,24 @@
   "exports": {
     ".": "./dist/index.js"
   },
+  "types": "./src/index.d.ts",
   "bin": {
     "gat": "bin/cli.js"
   },
   "files": [
     "bin",
-    "dist/cli.cjs"
+    "dist",
+    "src/index.d.ts"
   ],
   "scripts": {
-    "build": "esbuild src/cli.js --bundle --platform=node --target=node22 --format=cjs --minify --sourcemap --outfile=dist/cli.cjs",
+    "build": "npm run build:lib && npm run build:cli",
+    "build:cli": "esbuild src/cli.js --bundle --platform=node --target=node22 --format=cjs --minify --outfile=dist/cli.cjs",
+    "build:lib": "esbuild src/index.js --bundle --platform=node --target=node22 --format=esm --minify --sourcemap --outfile=dist/index.js",
     "format": "prettier --write .",
     "release": "semantic-release",
     "test": "jest --coverage --verbose",
     "test:all": "npm run test && npm run test:slow",
-    "pretest:slow": "npm run build",
+    "pretest:slow": "npm run build:cli",
     "test:slow": "jest --config jest.config.slow.js"
   },
   "dependencies": {

--- a/src/get.js
+++ b/src/get.js
@@ -7,19 +7,7 @@ import { createLog } from './log.js'
 import { getDownloadUrl, isArduinoTool, tools } from './tools.js'
 
 /**
- * @typedef {Object} GetToolParams
- * @property {import('./tools.js').Tool} tool
- * @property {string} version
- * @property {string} destinationFolderPath
- * @property {NodeJS.Platform} [platform=process.platform]
- * @property {NodeJS.Architecture} [arch=process.arch]
- * @property {boolean} [force=false]
- *
- * @typedef {Object} GetToolResult
- * @property {string} toolPath
- *
- * @param {GetToolParams} params
- * @returns {Promise<GetToolResult>}
+ * @type {typeof import('./index.js').getTool}
  */
 export async function getTool({
   tool,

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -1,0 +1,25 @@
+const arduinoTools = [
+  'arduino-cli',
+  'arduino-language-server',
+  'arduino-fwuploader',
+] as const
+const clangTools = ['clangd', 'clang-format'] as const
+
+export const tools = [...arduinoTools, ...clangTools]
+
+export declare type Tool = typeof tools[number]
+
+export declare interface GetToolParams {
+  tool: Tool
+  version: string
+  destinationFolderPath: string
+  platform?: NodeJS.Platform
+  arch?: NodeJS.Architecture
+  force?: boolean
+}
+
+export declare interface GetToolResult {
+  toolPath: string
+}
+
+export declare function getTool(params: GetToolParams): Promise<GetToolResult>

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -7,7 +7,7 @@ const clangTools = ['clangd', 'clang-format'] as const
 
 export const tools = [...arduinoTools, ...clangTools]
 
-export declare type Tool = typeof tools[number]
+export declare type Tool = (typeof tools)[number]
 
 export declare interface GetToolParams {
   tool: Tool

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -7,7 +7,9 @@ const clangTools = ['clangd', 'clang-format'] as const
 
 export const tools = [...arduinoTools, ...clangTools]
 
-export declare type Tool = (typeof tools)[number]
+export declare type ArduinoTool = (typeof arduinoTools)[number]
+export declare type ClangTool = (typeof clangTools)[number]
+export declare type Tool = ArduinoTool | ClangTool
 
 export declare interface GetToolParams {
   tool: Tool

--- a/src/tools.js
+++ b/src/tools.js
@@ -28,7 +28,7 @@ export function isArduinoTool(tool) {
 
 /**
  * @typedef {Object} GetDownloadUrlParams
- * @property {import('./index.js').Tool} tool
+ * @property {Tool} tool
  * @property {string} version
  * @property {NodeJS.Platform} platform
  * @property {NodeJS.Architecture} arch

--- a/src/tools.js
+++ b/src/tools.js
@@ -2,23 +2,33 @@ import { posix } from 'node:path'
 
 import { createLog } from './log.js'
 
-const arduinoTools = /** @type {const} */ ([
+/**
+ * @typedef {import('./index.js').Tool} Tool
+ * @typedef {import('./index.js').ArduinoTool} ArduinoTool
+ */
+
+const arduinoTools = [
   'arduino-cli',
   'arduino-language-server',
   'arduino-fwuploader',
+]
+const clangTools = ['clangd', 'clang-format']
+export const tools = /** @type {readonly Tool[]} */ ([
+  ...arduinoTools,
+  ...clangTools,
 ])
-const clangTools = /** @type {const} */ (['clangd', 'clang-format'])
-export const tools = /** @type {const} */ ([...arduinoTools, ...clangTools])
 
+/**
+ * @param {Tool} tool
+ * @returns {tool is ArduinoTool}
+ */
 export function isArduinoTool(tool) {
   return arduinoTools.includes(tool)
 }
 
 /**
- * @typedef {typeof tools[number]} Tool
- *
  * @typedef {Object} GetDownloadUrlParams
- * @property {Tool} tool
+ * @property {import('./index.js').Tool} tool
  * @property {string} version
  * @property {NodeJS.Platform} platform
  * @property {NodeJS.Architecture} arch

